app/routers/profile.py

```diff
@@ -148,3 +149,64 @@ def get_image_crop_area(width, height):
         return (delta, 0, width - delta, height)
     delta = (height - width) // 2
     return (0, delta, width, width + delta)
+
+
+@router.post("/update_holidays")
+async def update_holidays(
+        file: UploadFile = File(...), session=Depends(get_db)):
+    icsfile = await file.read()
+    holidays = get_holidays_from_file(icsfile.decode(), session)
+    try:
+        save_holidays_to_db(holidays, session)
+    except SQLAlchemyError as ex:
+        logger.error(ex)
+    finally:
+        url = router.url_path_for("profile")
+        return RedirectResponse(url=url, status_code=HTTP_302_FOUND)
+
+
+def get_holidays_from_file(file, session):
+    """
+    this function using regex to extract holiday title
+    and date from standrd ics file
+    :param file:standard ics file
+    :param session:current connection
+    :return:list of holidays events
+    """
+    parsed_holidays = REGEX_EXTRACT_HOLIDAYS.finditer(file)
+    holidays = []
+    for holiday in parsed_holidays:
+        valid_ascii_chars_range = 128
+        title = holiday.groupdict()['title'].strip()
+        title_to_save = ''.join([i if ord(i) < valid_ascii_chars_range
+                                 else '' for i in title])
+        date = holiday.groupdict()['date'].strip()
+        holiday = Event(
+            title=title_to_save,
+            start=datetime.strptime(date, '%Y%m%d'),
```

Comment:
Maybe save the format as a constat?

Reply:
saved it as a variable as its the only function which use it (for now). thanks.